src/nvim/os/shell.c

```diff
@@ -70,6 +73,24 @@ char **shell_build_argv(const char *cmd, const char *extra_args)
   return rv;
 }
 
+/// Builds the argument vector for any command with arguments.
+///
+/// Creates: ["command", "that", "has", "spaces"] from a command string
+///
+/// @param cmd Command string
+/// @return A newly allocated argument vector. It must also be freed with
+///         `shell_free_argv` when no longer needed.
+char **build_argv(char *cmd)
+  FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC FUNC_ATTR_NONNULL_ALL
+{
+  size_t argc = tokenize((char_u *)cmd, NULL);
+  char **rv = xmalloc((argc + 4) * sizeof(*rv));
```

Comment:
`4` -> `1`. We only need a pointer for every argument and the terminating `NULL` pointer.